pull.go

```diff
@@ -63,19 +63,46 @@ func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (_ Ima
 	}
 	defer done(ctx)
 
-	var unpacks int32
-	var unpackEg *errgroup.Group
-	var unpackWrapper func(f images.Handler) images.Handler
+	var unpacker *unpack.Unpacker
 
 	if pullCtx.Unpack {
-		// unpacker only supports schema 2 image, for schema 1 this is noop.
-		u, err := c.newUnpacker(ctx, pullCtx)
+		snapshotterName, err := c.resolveSnapshotterName(ctx, pullCtx.Snapshotter)
 		if err != nil {
-			return nil, fmt.Errorf("create unpacker: %w", err)
+			return nil, fmt.Errorf("unable to resolve snapshotter: %w", err)
+		}
+		var uconfig UnpackConfig
+		for _, opt := range pullCtx.UnpackOpts {
+			if err := opt(ctx, &uconfig); err != nil {
+				return nil, err
+			}
+		}
+		var platformMatcher platforms.Matcher
+		if !uconfig.CheckPlatformSupported {
+			platformMatcher = platforms.All
```

Comment:
the `WithUnpackPlatform` will apply `platforms.All` if the unpack.Platform.Platform is nil. It looks like that `CheckPlatformSupported` option is short circuit and useless.

Reply:
The logic here should be the same, but agreed it is kind of odd. I think it will be clearer though in the future having unpack platforms be explicit and separate from the fetcher's platform matcher if there is reason for them not to be the same.